src/api.js

```diff
@@ -59,3 +63,73 @@ api.declare({
   );
 });
 
+api.declare({
+/*Gets the namespace, creates one if one doesn't exist*/
+  method:   'get',
+  route:    '/namespace/:namespace',
+  name:     'namespace',
+  title:    'Create a namespace',	
+  scopes:   [
+    ['pulse:namespace:<namespace>'],
+  ],
+  //todo later: deferAuth: true,
+  description: [
+    'Creates a namespace, given the taskcluster credentials with scopes.',
+    '',
+    '**Warning** this api end-point is **not stable**.',
+  ].join('\n'),
+}, async function(req, res) {
+ 
+  let {namespace} = req.params;
+
+  await this.Namespaces.ensureTable(); //set up the table if needed
```

Comment:
This should be in the `main.js` setup for the `Namespaces` object rather than in this function because it only needs to be done once at startup, rather than on every request.